### PR TITLE
feat: base58 message import

### PR DESCRIPTION
**Pull request by joeymeere**
This PR adds the ability to paste a base58 serialized transaction message, simulate it, and propose it. This is done on the transaction page with a button across from the heading. ~This implementation does not support versioned messages, and by proxy address lookup tables. The imported transaction must be a standard `TransactionMessage` type.~

### CODE REVIEW

File: lib/transaction/decodeAndDeserialize.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as bs58 from "bs58";
+import { VersionedMessage, VersionedTransaction } from "@solana/web3.js";
+
+export function decodeAndDeserialize(tx: string): VersionedMessage {
+  try {
+    const messageBytes = bs58.default.decode(tx);
+    return VersionedMessage.deserialize(messageBytes);
```
Comment (bguillaumat) [inline lib/transaction/decodeAndDeserialize.ts:7]: The "issue" is that here we rely on the fact its only a versioned tx

Comment (joeymeere) [reply lib/transaction/decodeAndDeserialize.ts]: updated in [728c8e2](https://github.com/Squads-Protocol/squads-v4-public-ui/pull/11/commits/728c8e223eb16ef320267100b78c3a5330ffc379) to handle both types

**Review by bguillaumat — APPROVED** [review]
utACK

@joeymeere you did tested with Legacy and Versioned?

**Comment by joeysqds** [comment]
> utACK
> 
> @joeymeere you did tested with Legacy and Versioned?

tested with both

**Pull request merged**